components/brave_wallet/browser/solana_instruction_data_decoder.h

```diff
@@ -15,6 +15,7 @@
 
 namespace brave_wallet::solana_ins_data_decoder {
 
+absl::optional<GURL> DecodeMetadataUri(const std::vector<uint8_t> data);
```

Comment:
This doesn't seem like the perfect place for this function (since it's not a Solana instruction).  Is there a better place? Or maybe we should generalize such that solana_instruction_data_decoder becomes solana_data_decoder?

Reply:
You can move this to solana_data_decoder_util and other Decode* functions.

Reply:
Updated. Please let me know if I separated things the way you were expecting.